controller/admin_controller.php

```diff
@@ -0,0 +1,296 @@
+	protected $phpbb_root_path;
+
+	/** @var string */
+	protected $php_ext;
+
+	/** @var array */
+	protected $cfg_array = array();
+
+	/** @var array */
+	protected $new_config = array();
+
+	/** @var string */
+	public $u_action;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\config\config              $config               Config object
+	 * @param \phpbb\db\driver\driver_interface $db                   Database object
+	 * @param \phpbb\language\language          $language             Language object
+	 * @param \phpbb\log\log                    $log                  Log object
+	 * @param \phpbb\request\request            $request              Request object
+	 * @param \phpbb\template\template          $template             Template object
+	 * @param \phpbb\user                       $user                 User object
+	 * @param string                            $phpbb_root_path      phpBB root path
+	 * @param string                            $php_ext              php_ext
+	 * @access public
+	 */
+	public function __construct(\phpbb\config\config $config, \phpbb\db\driver\driver_interface $db, \phpbb\language\language $language, \phpbb\log\log $log, \phpbb\request\request $request, \phpbb\template\template $template, \phpbb\user $user, $root_path, $php_ext)
+	{
+		$this->config = $config;
+		$this->db = $db;
+		$this->language = $language;
+		$this->log = $log;
+		$this->request = $request;
+		$this->template = $template;
+		$this->user = $user;
+		$this->phpbb_root_path = $root_path;
+		$this->php_ext = $php_ext;
+	}
+
+	/**
+	 * Display the options a user can configure for this extension
+	 *
+	 * @return null
+	 * @access public
+	 */
+	public function display_options()
+	{
+		// Create a form key for preventing CSRF attacks
+		add_form_key('acp_phpbb_ideas_settings');
+
+		$this->new_config = $this->config;
+		$this->cfg_array = ($this->request->is_set('config')) ? $this->request->variable('config', array('' => ''), true) : $this->new_config;
+
+		// Check if selected user exists
+		$this->check_ideas_topics_poster();
+
+		// Configuration options to list through
+		$display_vars = array(
+			'ideas_forum_id',
+			'ideas_poster_id',
+			'ideas_base_url',
+			'ideas_forum_setup',
+		);
+		$this->set_config_options($display_vars);
+
+		// Output relevant page
+		$this->template->assign_vars(array(
+			'IDEAS_POSTER'		=> $this->get_ideas_topics_poster(),
+			'IDEAS_BASE_URL'	=> ($this->config['ideas_base_url']) ?: '',
+
+			'S_FORUM_SELECT_BOX'	=> $this->select_ideas_forum(),
+			'S_IDEAS_FORUM_ID'		=> !empty($this->config['ideas_forum_id']),
+
+			'U_ACTION'			=> $this->u_action,
+			'U_FIND_USERNAME'	=> append_sid("{$this->phpbb_root_path}memberlist.{$this->php_ext}", 'mode=searchuser&amp;form=acp_phpbb_ideas_settings&amp;field=ideas_poster_id&amp;select_single=true'),
+		));
+	}
+
+	/**
+	 * Check if Ideas bot user exists
+	 *
+	 * @return null
+	 * @access public
+	 */
+	public function check_ideas_topics_poster()
```

Comment:
I would even rename this to something like get_user or get_user_by_name or something similar.